src/cpp/core/system/PosixSystem.cpp

```diff
@@ -2403,8 +2403,28 @@ Error userBelongsToGroup(const User& user,
                          bool* pBelongs)
 {
    *pBelongs = false; // default to not found
+
+   // first we try to use getgrouplist(3) to determine if the user is a
+   // member of the target groupName
+   std::vector<group::Group> groups;
+   Error error = group::userGroups(user.getUsername(), &groups);
+   if (error)
+      return error;
```

Comment:
Is it possible that getgrouplist(3) could fail, but the following getgrnam_r(3) call could succeed? If so, it seems to me like we should save this error and return it later only if the following attempt to retrieve group names fails.

It might also be worthwhile to have two separate functions which are called + used here, e.g. `userBelongsToGroupViaGroupList()` and `userBelongsToGroupViaGroupName()`.